### PR TITLE
ITNFA-30: Extended results of findByBiometricMatch by MPI patients

### PR DESCRIPTION
**Pull request by dserkowski**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/registrationcore/RegistrationCoreConstants.java [hunk 1]
```diff
@@ -96,4 +96,8 @@
 	public static final String GP_ERROR_HANDLER_IMPLEMENTATION = "registrationcore.errorHandler.implementation";
 
 	public static final String MPI_IDENTIFIER_TYPE_ECID_NAME = "ECID";
-}
\ No newline at end of file
+
+	public static final String LOCAL_FINGERPRINT = "localFingerprint";
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/registrationcore/RegistrationCoreConstants.java]: I would change it to LOCAL_FINGERPRINT_NAME or key to emphasize that it doesn't contain some discrete value (see MPI_IDENTIFIER_TYPE_ECID_NAME)

File: api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java [hunk 2]
```diff
@@ -462,10 +469,17 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
 					log.warn("Found " + patients.size() + " patients with the same biometric id: " + subjectId);
 				}
 				for (Patient patient : patients) {
+					possessNationalFpId = possessNationalFpId(patient) || possessNationalFpId;
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java]: This line looks quite confusing. I think we should avoid the same parameter and method names.
Moreover, if it is possible we can use Java 8 stream with findAny() method which allows to determine if any element possess national fp id with more 'clean code' way

File: api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java [hunk 3]
```diff
@@ -462,10 +469,17 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
...
 			}
+
+			if (registrationCoreProperties.isMpiEnabled() && !possessNationalFpId) {
+				Patient mpiPatient = findMpiPatient(subjectId, nationalBiometricId.getUuid());
+				result.add(new PatientAndMatchQuality(mpiPatient,
+						match.getMatchScore(), Collections.singletonList(NATIONAL_FINGERPRINT)));
+			}
 		}
 
 		return result;
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java]: findByBiometricMatch could be split into smaller methods with the descriptive method names. However no changes required, just a hint.

**Review by pawelcieszko — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java (hunk 2 above)
Comment (dserkowski) [reply api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java]: In this module we don't have Java 8

File: api/src/main/java/org/openmrs/module/registrationcore/RegistrationCoreConstants.java (hunk 1 above)
Comment (dserkowski) [reply api/src/main/java/org/openmrs/module/registrationcore/RegistrationCoreConstants.java]: Right, I will change it

File: api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java (hunk 2 above)
Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java]: Ok, however parameter name and method name shouldn't be the same

**Review by pawelcieszko — APPROVED** [review]

**Pull request merged**